pkg/controllers/provisioning/scheduling/preferences.go

```diff
@@ -36,28 +36,27 @@ type Preferences struct {
 }
 
 func (p *Preferences) Relax(ctx context.Context, pod *v1.Pod) bool {
-	ctx = log.IntoContext(ctx, log.FromContext(ctx).WithValues("Pod", klog.KRef(pod.Namespace, pod.Name)))
 	relaxations := []func(*v1.Pod) *string{
-		p.removeRequiredNodeAffinityTerm,
-		p.removePreferredPodAffinityTerm,
-		p.removePreferredPodAntiAffinityTerm,
-		p.removePreferredNodeAffinityTerm,
-		p.removeTopologySpreadScheduleAnyway}
+		p.RemoveRequiredNodeAffinityTerm,
+		p.RemovePreferredPodAffinityTerm,
+		p.RemovePreferredPodAntiAffinityTerm,
+		p.RemovePreferredNodeAffinityTerm,
+		p.RemoveTopologySpreadScheduleAnyway}
 
 	if p.ToleratePreferNoSchedule {
-		relaxations = append(relaxations, p.toleratePreferNoScheduleTaints)
+		relaxations = append(relaxations, p.ToleratePreferNoScheduleTaints)
 	}
 
 	for _, relaxFunc := range relaxations {
 		if reason := relaxFunc(pod); reason != nil {
-			log.FromContext(ctx).V(1).Info(fmt.Sprintf("relaxing soft constraints for pod since it previously failed to schedule, %s", lo.FromPtr(reason)))
+			log.FromContext(ctx).WithValues("Pod", klog.KRef(pod.Namespace, pod.Name)).V(1).Info(fmt.Sprintf("relaxing soft constraints for pod since it previously failed to schedule, %s", lo.FromPtr(reason)))
 			return true
 		}
 	}
 	return false
 }
 
-func (p *Preferences) removePreferredNodeAffinityTerm(pod *v1.Pod) *string {
+func (p *Preferences) RemovePreferredNodeAffinityTerm(pod *v1.Pod) *string {
```

Comment:
Why export all these functions?

Reply:
Good call -- made the update so that these aren't exported anymore. Forgot that everything here is happening in the same package